### PR TITLE
Solved Issue: #6 Baseline Model Implementation

### PR DESCRIPTION
**Pull request by simon2326**
# Solved Issue: #6  Baseline Model Implementation

## ✨ Context

This PR resolves issue #6 by implementing a baseline model for heart disease prediction. The goal is to establish a simple yet informative starting point that serves as a reference for evaluating future machine learning models.

## 🧠 Rationale behind the change

The baseline model provides a heuristic approach based on key insights from the Exploratory Data Analysis (EDA). This model is important for:

- Establishing a benchmark for predictive performance.
- Comparing future models against a simple rule-based approach.
- Identifying early strengths and weaknesses in feature selection.

## Type of changes

- [x] ✨ New Feature
- [x] 💥 Breaking Change
- [x] ⚗️  Experiments
- [x] 📝 Documentation

## 🛠 What does this PR implement

- Added `BaselineModel.ipynb`, which develops a heuristic model for heart disease prediction.
- Defined a set of heuristic rules based on EDA findings, such as: Patients with asymptomatic chest pain have a high likelihood of heart disease.
- Preprocessed data using scikit-learn pipelines.
- Implemented a custom heuristic model using sklearn.
- Evaluated the model using cross-validation, reporting accuracy, precision, **recall**, and F1-score.
- Documented findings and key insights from the baseline model performance and recall metric (chosen one).

## 🧪 How should this be tested?

- Run `BaselineModel.ipynb`.
- Verify data preprocessing and feature selection.
- Ensure the heuristic model runs correctly and produces predictions.
- Review evaluation metrics (accuracy, precision, **recall**, F1-score).
- Confirm that insights align with previous EDA findings.

## Analysis of Results

- The high recall (97.2%) suggests that the model correctly identifies most patients with heart disease.
- The lower precision (54.6%) means there are many false positives.
- The accuracy (55.4%) is relatively low but expected for a rule-based approach.
- This model prioritizes recall to minimize false negatives, ensuring that most heart disease cases are detected.

### CODE REVIEW

**Comment by simon2326** [comment]
@JoseRZapata @Susanagg05 @alejolondonm @1210-lof @spuertaf2002 

`Check my PR please`

**Review by alejolondonm — APPROVED** [review]

File: BaselineModel.ipynb [hunk 1]
```diff
@@ -0,0 +1,1870 @@
...
+    "\n",
+    "nom_categorical_pipe = Pipeline(\n",
+    "    steps=[\n",
+    "        (\"imputer\", SimpleImputer(strategy=\"most_frequent\")),\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "boolean_pipe = Pipeline(\n",
+    "    steps=[\n",
+    "        (\"to_int\", FunctionTransformer(lambda x: x.astype(\"int64\"), feature_names_out=\"one-to-one\")),\n",
```
Comment (spuertaf2002) [inline BaselineModel.ipynb:490]: Love this!

File: BaselineModel.ipynb [hunk 2]
```diff
@@ -0,0 +1,1870 @@
...
+    "#### **Pipeline**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 98,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.preprocessing import FunctionTransformer\n",
```
Comment (spuertaf2002) [inline BaselineModel.ipynb:474]: @simon2326 Please keep all imports at the start of the file. It's the better practice.

File: BaselineModel.ipynb [hunk 3]
```diff
@@ -0,0 +1,1870 @@
...
+    "\n",
+    "numeric_pipe = Pipeline(\n",
+    "    steps=[\n",
+    "        (\"imputer\", SimpleImputer(strategy=\"median\")),\n",
```
Comment (spuertaf2002) [inline BaselineModel.ipynb:478]: @simon2326 Have caution with this, imputing max_hr only with the median it's going to shift significantly (by 23%) the distribution. Consider using more robust techniques as the KNN imputer.
Dataset consists of 474 records, 363 of those for max_hr are non NAN values, leaving us with 111 NAN values. Those 111 NAN represent (111 / 474) * 100 = 23.417% of NAN for the column.

Comment (spuertaf2002) [reply BaselineModel.ipynb:478]: @JoseRZapata Is it fair to request file changes do to this concern? Love to know your take on this one!

Comment (JoseRZapata) [reply BaselineModel.ipynb:478]: This is a first approach to the feature engineering so more experiments are needed to figure out the impact of this simple imputer.
But I prefer a reasonable simple solution in the first iteration. So its Fine, in this academic situation.

Comment (spuertaf2002) [reply BaselineModel.ipynb:478]: @JoseRZapata Thanks for the reply!

**Review by spuertaf2002 — APPROVED** [review]
All LGTM!

File: BaselineModel.ipynb (hunk 1 above)
Comment (simon2326) [reply BaselineModel.ipynb:490]: Thx!!

File: BaselineModel.ipynb (hunk 2 above)
Comment (simon2326) [reply BaselineModel.ipynb:474]: I need to work on the good practices, thanks for letting me know that.

File: BaselineModel.ipynb (hunk 3 above)
Comment (simon2326) [reply BaselineModel.ipynb:478]: I agree with you Santiago, you're right, I should've checked another method for those lots of nan values.

**Review by JoseRZapata — APPROVED** [review]
Simon muy buen trabajo,  se cumple el objetivo de la tarea, y es muy claro como realizaste el procedimiento de feature engineering. y el modelo baseline.

En esta tarea se le daba prioridad a una métrica pero lo mejor siemrpe es tomar varias y buscar un buen balance entre todas. Los ersulados del recall son muy diferentes a las otras, entonces el sistema esta un poco descompensado.

Excelente trabajo.

**Comment by simon2326** [comment]
> Simon muy buen trabajo, se cumple el objetivo de la tarea, y es muy claro como realizaste el procedimiento de feature engineering. y el modelo baseline.
> 
> En esta tarea se le daba prioridad a una métrica pero lo mejor siemrpe es tomar varias y buscar un buen balance entre todas. Los ersulados del recall son muy diferentes a las otras, entonces el sistema esta un poco descompensado.
> 
> Excelente trabajo.

Gracias Profe por tu retroalimentación, tienes toda la razón!
